src/coreclr/System.Private.CoreLib/src/System/Threading/Monitor.CoreCLR.cs

```diff
@@ -68,8 +146,35 @@ private static void ThrowLockTakenException()
         **             SynchronizationLockException if the current thread does not
         **             own the lock.
         =========================================================================*/
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern void Exit(object obj);
+        public static void Exit(object obj)
+        {
+            LeaveHelperAction exitBehavior = Exit_FastPath(obj);
+
+            if (exitBehavior == LeaveHelperAction.None)
+                return;
+
+            Exit_Slowpath(obj, exitBehavior);
+        }
+
+        // Used to implement synchronized methods on non Windows-X86 architectures
+        internal static void ExitIfLockTaken(object obj, ref bool lockTaken)
+        {
+            ArgumentNullException.ThrowIfNull(obj);
+
+            if (lockTaken)
+            {
+                LeaveHelperAction exitBehavior = Exit_FastPath(obj);
+
+                if (exitBehavior == LeaveHelperAction.None)
+                {
+                    lockTaken = false;
+                    return;
+                }
+
+                Exit_Slowpath(obj, exitBehavior);
```

Comment:
Can `Exit_FastPath` do anything when returning `exitBehavior != LeaveHelperAction.None` (note _NOT EQUAL_) that would break the subsequent fall-through to `Exit_Slowpath` (e.g. we did the fast-path exit, but it needs some other LeaveHelperAction... but we've done the Exit already)?

(same question on line 154)

Reply:
No.